Python/libraries/recognizers-text/recognizers_text/matcher/string_matcher.py

```diff
@@ -41,42 +40,33 @@ def matcher(self) -> Matcher:
     def matcher(self, matcher) -> None:
         self.__matcher = matcher
 
-    @dispatch(list)
-    def init(self, values: []) -> None:
-        self.init(values, list(map(lambda v: str(v), values)))
-
-    @dispatch(dict)
-    def init(self, values_dictionary: {}) -> None:
-        values = []
-        ids = []
-        for item in values_dictionary:
-            id = item
-            for value in values_dictionary[item]:
-                values.append(value)
-                ids.append(id)
-
-        self.init(values, ids)
-
-    @dispatch(list, list)
-    def init(self, values, ids: [] = []) -> None:
-        tokenized_values = self.get_tokenized_text(values)
-        self.matcher.init(tokenized_values, ids)
-
-    @dispatch(list)
-    def find(self, tokenized_query: []) -> []:
-        return self.matcher.find(tokenized_query)
-
-    @dispatch(str)
-    def find(self, query_text: str = "") -> []:
-        query_tokens = self.__tokenizer.tokenize(query_text)
+    def init(self, values, ids: [] = None):
+        if values and type(values) is list:
+            _ids = list(map(lambda v: str(v), values)) if not ids else ids
+            tokenized_values = self.get_tokenized_text(values)
+            self.matcher.init(tokenized_values, _ids)
+        elif values and type(values) is dict:
+            values_dictionary = []
+            for item in values_dictionary:
+                id = item
+                for value in values[item]:
+                    values_dictionary.append(value)
+                    ids.append(id)
+
+            self.init(values, ids)
+
```

Comment:
```
def init(self, values, ids: [] = None):
    if isinstance(values, dict):
        values_dictionary = []
        for item in values_dictionary:
            id = item
            for value in values[item]:
                values_dictionary.append(value)
                ids.append(id)
    
    _ids = list(map(lambda v: str(v), values)) if not ids else ids
    tokenized_values = self.get_tokenized_text(values)
    self.matcher.init(tokenized_values, _ids)
```

I think this would work and you'd save up some time by not calling the function again